src/libraries/Common/src/System/Marvin.cs

```diff
@@ -53,19 +53,47 @@ private static unsafe long ComputeHash(ref byte rBuffer, nuint cbBuffer, uint p0
                 switch ((int)cbBuffer & 3)
                 {
                     case 0:
-                        p0 += 0x80u;
+                        if (BitConverter.IsLittleEndian)
+                        {
+                            p0 += 0x80u;
+                        }
+                        else
+                        {
+                            p0 += 0x80000000u;
+                        }
                         break;
 
                     case 1:
-                        p0 += 0x8000u | pbBuffer[currentOffset];
+                        if (BitConverter.IsLittleEndian)
+                        {
+                            p0 += 0x8000u | pbBuffer[currentOffset];
+                        }
+                        else
+                        {
+                            p0 += 0x800000u | ((uint)pbBuffer[currentOffset] << 24);
+                        }
                         break;
 
                     case 2:
-                        p0 += 0x800000u | Unsafe.ReadUnaligned<ushort>(pbBuffer + currentOffset);
+                        if (BitConverter.IsLittleEndian)
+                        {
+                            p0 += 0x800000u | Unsafe.ReadUnaligned<ushort>(pbBuffer + currentOffset);
+                        }
+                        else
+                        {
+                            p0 += 0x8000u | ((uint)Unsafe.ReadUnaligned<ushort>(pbBuffer + currentOffset) << 16);
+                        }
                         break;
 
                     case 3:
-                        p0 += 0x80000000u | Unsafe.ReadUnaligned<ushort>(pbBuffer + currentOffset) | ((uint)pbBuffer[currentOffset + 2] << 16);
+                        if (BitConverter.IsLittleEndian)
+                        {
+                            p0 += 0x80000000u | Unsafe.ReadUnaligned<ushort>(pbBuffer + currentOffset) | ((uint)pbBuffer[currentOffset + 2] << 16);
+                        }
+                        else
+                        {
+                            p0 += 0x80u | ((uint)Unsafe.ReadUnaligned<ushort>(pbBuffer + currentOffset) << 16) | ((uint)pbBuffer[currentOffset + 2] << 8);
+                        }
```

Comment:
I think the code would read better if the whole switch block were under a single if/else rather than having one per case.

Reply:
I just realized this whole Marvin class is a duplicate, and it's used only by IPAddress' GetHashCode.  I think we should just delete this whole file and not use it from IPAddress.  I'll look at doing so.

Reply:
OK, I'll take this change of the patch then.  Thanks!